blocklist_user.c

```diff
@@ -0,0 +1,90 @@
+#include <linux/netlink.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#define MAX_PAYLOAD 1024
+#define LINE_LENGTH 20
+
+int main(int argc, char *argv[])
+{
+    /* Read interface pair from blocklist.txt */
+    if (argc != 2) {
+        printf("Error: Unspecified file name\n");
+        exit(1);
+    }
+    char *filename = argv[1];
+    FILE *fp = fopen(filename, "r");
```

Comment:
Instead of reading from a files, let's utilize command line option to specify the rules for blocking.
You would need [getopt](https://man7.org/linux/man-pages/man3/getopt.3.html).